src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionService.cs

```diff
@@ -192,11 +186,11 @@ protected virtual void GetComponentDesignerActions(IComponent component, Designe
                         DesignerVerbCollection verbs = dcs.Verbs;
                         if (verbs is not null && verbs.Count != 0)
                         {
-                            ArrayList verbsArray = new ArrayList();
-                            bool hookupEvents = _componentToVerbsEventHookedUp[component] is null;
+                            List<DesignerVerb> verbsArray = new();
+                            bool hookupEvents = !_componentToVerbsEventHookedUp.Contains(component);
```

Comment:
```suggestion
                            bool hookupEvents = _componentToVerbsEventHookedUp.Add(component);
```
and delete `if` below?

Reply:
Thanks, done.